src/features/shared-link-settings-modal/SharedLinkSettingsModal.js

```diff
@@ -212,8 +214,8 @@ class SharedLinkSettingsModal extends Component {
         this.setState({ isPasswordEnabled: event.target.checked });
     };
 
-    onExpirationDateChange = date => {
-        this.setState({ expirationDate: date, expirationError: undefined });
+    onExpirationDateChange = (date, formattedDate) => {
+        this.setState({ expirationDate: new Date(formattedDate), expirationError: undefined });
```

Comment:
could we confirm that existing uses are unaffected by this change?

does this work as expected for the different `dateFormat` options? (i.e. are all the `formattedDate` values valid to be passed into `new Date()`?)